tests/test_empirical_orthogonal_functions.py

```diff
@@ -603,7 +603,7 @@ def test_EOFDataForAllDOYs_alldoy_getfunctions():
         eofs.append(eof.EOFData(lat, long, eof1, eof2,
                                 explained_variances=np.squeeze(explained_variances[:, doy - 1]),
                                 eigenvalues=np.squeeze(eigenvalues[:, doy - 1]), no_observations=no_obs[doy - 1]))
-    target = eof.EOFDataForAllDOYs(eofs, no_leap)
+    target = eof.EOFDataForAllDOYs(eofs, no_leap_years)
```

Comment:
It appears that the no_leap_years setting is not checked below, right?

This should be added in the test, I guess, although not much can go wrong. However, this tests basically tests that the things, which only rarely  go wrong really don't go wrong. Maybe you could check if analogous situations occur in the other tests here?

Reply:
added several tests for no_leap_years